pkg/configfilearg/parser.go

```diff
@@ -110,31 +111,69 @@ func (p *Parser) findStart(args []string) ([]string, []string, bool) {
 	return args, nil, false
 }
 
+func dotDFiles(basefile string) (result []string, _ error) {
+	files, err := ioutil.ReadDir(basefile + ".d")
+	if os.IsNotExist(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+	for _, file := range files {
+		if file.IsDir() || (!strings.HasSuffix(file.Name(), ".yaml") && !strings.HasSuffix(file.Name(), ".yml")) {
```

Comment:
import `"github.com/rancher/k3s/pkg/agent/util"` and do:
```suggestion
		if file.IsDir() || !util.HasSuffixI(file.Name(), ".yaml", ".yml") {
```

Reply:
Updated